provision/docker/provisioner.go

```diff
@@ -1008,6 +1008,19 @@ func (p *dockerProvisioner) ListNodes(addressFilter []string) ([]provision.Node,
 	return result, nil
 }
 
+func (p *dockerProvisioner) ListNodesByFilter(filter map[string]string) ([]provision.Node, error) {
+	nodes, err := p.Cluster().UnfilteredNodesForMetadata(filter)
+	if err != nil {
+		return nil, err
+	}
+	result := make([]provision.Node, 0, len(nodes))
+	for i := range nodes {
+		n := &nodes[i]
+		result = append(result, &clusterNodeWrapper{Node: n, prov: p})
```

Comment:
Since you know the slice's capacity is enough, you can assign the data directly to their positions.

```suggestion
		result[i] = &clusterNodeWrapper{Node: n, prov: p}
```

Reply:
Thanks, @nettoclaudio ! I've already done the suggested changes.